### ISSUE TITLE
Tesseract merge?

### ISSUE THREAD

**Issue by aolko** [inexor-game/code#164]
How about merging with http://tesseract.gg?

**Comment by a-teammate** [comment]
Seems legit.
Imo it should be merged (maybe as big step for the second final release), but i'd rather like to use it as a prototype and reuse @eihrul's changes on tesseract instead of merging it forcefully.
Would be a shitload of work, but maybe we can bring it to work in parallel to the old lightmap system.
We should at least keep that in mind designing the next mapformat (i.e. light entities of different versions)..
I think a lot of nice stuff already can be adopted now. E.g. multiplayer undo and multiplayer vslots, #114. 
And the various code cleanups ofc also could help.

**Comment by aschaeffer** [comment]
We have been asked several times about merging Tesseract and in particular the rendering subsystem. There are other improvements as well, but I'll focus on this topic.

The most important technology introduced in Tesseract is deferred shading which solves the problem of having performance issues with many lights. Whereas the Cube 2 engine calculates lightmaps in beforehand which can take hours on a complex map, Tesseract does it in realtime (which means at least 60 fps).

I think it IS a huge task and we are currently working on other huge tasks (HTML5 rendering engine, new entity system, new particle system, new map format, improved content packaging). Although I would like to see deferred shading in Inexor, we should wait until we finished the current work.

**Comment by SF-Fatality** [comment]
Agree, agree, and agree.

**Comment by Croydon** [comment]
I agree that it sounds a lot more sophisticated and stable when we merge features of Tesseract step-by-step over time.

**Comment by aschaeffer** [comment]
Some features like the one mentioned by @a-teammate can be merged step by step, but I think merging the deferred shading rendering is one monolithic block of work.

Also, I think we wouldn't merge the deferred shading 1:1 (new entity system = new lights, new map format, ambient occlusion, mixing light maps and deferred shading).

**Comment by chasester** [comment]
I like how people think its like turning on a switch, tesseract code is immensely different from cube 2:sauerbraten and therefore way different from inexor. There is about 4000+ lines of code that tesseract adds to the project, not counting all the code changes. Its not like an external library that can be just replaced.

@a-teammate 
IDK if you can use lightmaps with the realtime lights, Tho the concept sounds interesting, using light baking, to control static lights to improve improving, Tho I dont think the old light map code can actually work as a bake map.

**Comment by Croydon** [comment]
Closed in favor of #310.
